src/nbl/video/utilities/CScanner.cpp

```diff
@@ -19,24 +18,59 @@ core::smart_refctd_ptr<asset::ICPUShader> CScanner::createShader(const bool indi
 		};
 
 		if(indirect)
-			glsl = loadBuiltinData("nbl/builtin/glsl/scan/indirect.comp");
+			hlsl = loadBuiltinData("nbl/builtin/hlsl/scan/indirect.hlsl");
 		else
-			glsl = loadBuiltinData("nbl/builtin/glsl/scan/direct.comp");
+			hlsl = loadBuiltinData("nbl/builtin/hlsl/scan/direct.hlsl");
 	}
-	auto buffer = core::make_smart_refctd_ptr<asset::ICPUBuffer>(glsl->getSize());
-	memcpy(buffer->getPointer(), glsl->getMappedPointer(), glsl->getSize());
-	auto cpushader = core::make_smart_refctd_ptr<asset::ICPUShader>(std::move(buffer), asset::IShader::ESS_COMPUTE, asset::IShader::E_CONTENT_TYPE::ECT_GLSL, "????");
+	auto buffer = core::make_smart_refctd_ptr<asset::ICPUBuffer>(hlsl->getSize());
+	memcpy(buffer->getPointer(), hlsl->getMappedPointer(), hlsl->getSize());
+	auto cpushader = core::make_smart_refctd_ptr<asset::ICPUShader>(std::move(buffer), asset::IShader::ESS_COMPUTE, asset::IShader::E_CONTENT_TYPE::ECT_HLSL, "????");
+
+	// (REVIEW): All of the below, that rely on enumerations, should probably be changed to take advantage of HLSL-CPP compatibility.
+	// Issue is that CScanner caches all shaders for all permutations of scanType-dataType-operator and it's not clear how to 
+	// do this without enums
```

Comment:
don't worry about caching for the time being, I'd use template types instead of enums, cause we can.